symengine/polynomial_multivariate.h

```diff
@@ -0,0 +1,442 @@
+#ifndef SYMENGINE_POLYNOMIALS_MULTIVARIATE
+#define SYMENGINE_POLYNOMIALS_MULTIVARIATE
+
+#include <symengine/expression.h>
+#include <symengine/monomials.h>
+
+namespace SymEngine
+{
+
+class MultivariateIntPolynomial : public Basic
+{
+public:
+    // vars: set of variables for the polynomial
+    // degrees: max degrees of the symbols
+    // dict: dictionary for sparse represntation of polynomial, x**1 * y**2 + 3
+    // * x**4 * y ** 5
+    // is represented as {(1,2):1,(4,5):3}
+    set_sym vars_;
```

Comment:
Can you keep this as a `set_basic`? When we want to use generators, this will have to be changed to `set_basic`, so I think it'll be good to have the ability from the beginning.

From generators I mean `sin(x)` and others. For example, SymPy Polys can represent `x**2 + sin(x)**2` as a polynomial of `x` and `sin(x)`.